airflow/contrib/hooks/ftp_hook.py

```diff
@@ -226,7 +226,11 @@ def rename(self, from_name, to_name):
     def get_mod_time(self, path):
         conn = self.get_conn()
         ftp_mdtm = conn.sendcmd('MDTM ' + path)
-        return datetime.datetime.strptime(ftp_mdtm[4:], '%Y%m%d%H%M%S')
+        time_val = ftp_mdtm[4:]
+        #time_val optionally has microseconds
+        if "." in time_val:
```

Comment:
You should prefer try/except over if/else if that results in

- speed-ups (for example by preventing extra lookups)
- cleaner code (less lines/easier to read)

http://stackoverflow.com/questions/7604636/better-to-try-something-and-catch-the-exception-or-test-if-its-possible-first

From the Python docs:

> Easier to ask for forgiveness than permission. This common Python coding style assumes the existence of valid keys or attributes and catches exceptions if the assumption proves false. This clean and fast style is characterized by the presence of many  try and except statements. The technique contrasts with the LBYL style common to many other languages such as C.